apps/prairielearn/src/lib/course-request.ts

```diff
@@ -0,0 +1,104 @@
+import { loadSqlEquiv, queryRow, queryAsync } from '@prairielearn/postgres';
+import { z } from 'zod';
+import { logger } from '@prairielearn/logger';
+import * as Sentry from '@prairielearn/sentry';
+
+import { DateFromISOString, IdSchema } from '../lib/db-types';
+import { createCourseRepoJob } from '../lib/github';
+import { sendCourseRequestMessage } from '../lib/opsbot';
+
+const sql = loadSqlEquiv(__filename);
+
+const JobsRowSchema = z.object({
+  authn_user_id: IdSchema.optional(),
+  finish_date: DateFromISOString.optional(),
+  id: IdSchema.optional(),
+  number: z.number().optional(),
+  start_date: DateFromISOString.optional(),
+  status: z.string().optional(),
+});
+
+const CourseRequestRowSchema = z.object({
+  approved_by_name: z.string().nullable(),
+  first_name: z.string().nullable(),
+  github_user: z.string().nullable(),
+  id: IdSchema,
+  institution: z.string().nullable(),
+  jobs: JobsRowSchema,
```

Comment:
Check the query; this should actually be `z.array(JobsRowSchema)`. I think the query is actually slightly broken in its current form, which is why this worked. Specifically, I think you'll need to make this change:

```diff
-coalesce(j.jobs, '{}'::jsonb)
+coalesce(j.jobs, '[]'::jsonb)

Reply:
Now returning an array in the query and have updated the schema to reflect this in [f0ce24a](https://github.com/PrairieLearn/PrairieLearn/pull/9238/commits/f0ce24a1b18a423153c5e9d4bbb93ecdbfd65f29).